pandas/core/frame.py

```diff
@@ -3386,12 +3387,8 @@ class    max    type
             new_obj = self.copy()
 
         def _maybe_casted_values(index, labels=None):
-            if isinstance(index, PeriodIndex):
-                values = index.astype(object).values
-            elif isinstance(index, DatetimeIndex) and index.tz is not None:
-                values = index
-            else:
-                values = index.values
+            values = index._values
+            if not isinstance(index, (PeriodIndex, DatetimeIndex)):
```

Comment:
what are you trying to do here?

Reply:
This is identical to what happened before in the `else` branch. Seems like it's doing some type inference / coercion.